src/algebra/pointwise.lean

```diff
@@ -576,7 +576,7 @@ section monoid
 
 /-- An alias for `set α`, which has a semiring structure given by `∪` as "addition" and pointwise
   multiplication `*` as "multiplication". -/
-@[derive inhabited] def set_semiring (α : Type*) : Type* := set α
+@[derive [inhabited, partial_order, order_bot]] def set_semiring (α : Type*) : Type* := set α
```

Comment:
I think the linter errors are telling us something wonky is going on here. Does it behave any better if we use:
```lean
instance : partial_order (set_semiring α) :=
{ le := λ s t, s.down ⊆ t.down,
  lt := λ s t, s.down ⊂ t.down,
  ..(by apply_instance : set.partial_order) }
```

Reply:
I tried moving the `derive` to being explicit instances, but linting on my machine still complains about the `simp normal form` of the two `simp` lemmas.  I do not really know how to fix this, other than possibly removing the `simp` tag to these lemmas.

Reply:
Since the linter is unhappy with either attempt, I would revert to the version with the `derive`, no?

Reply:
Thanks for trying. How annoying...